app/main.dev.js

```diff
@@ -677,6 +764,50 @@ ipcMain.on('bluetooth-pairing-response', (event, response) => {
   bluetoothPinCallback(response);
 });
 
+ipcMain.on('native-message', (event, msg) => {
+  // Check if process needs to be restarted
+  if (!proc || proc.killed || !proc.stdin || proc.stdin.destroyed) {
+    console.log('[native-message] Helper process not available, restarting...');
+    initHelperProcess();
+
+    // Wait a bit for the process to start before sending the message
+    setTimeout(() => {
+      if (proc && !proc.killed && proc.stdin && !proc.stdin.destroyed) {
+        sendMessageToHelper(msg);
+      } else {
+        console.error('[write error] Failed to restart helper process');
+        // Send error back to renderer
+        if (mainWindow && !mainWindow.isDestroyed()) {
+          const error = {
+            msgType: 'error',
+            details: 'Helper process is not available'
+          };
+          mainWindow.webContents.send('native-reply', error);
+        }
+      }
+    }, 100);
```

Comment:
I'm slightly concerned that 100ms might not be sufficient for thin-client class machines that often end up being used in clinical settings. Would some sort of polling with a TTL be better than a one-time hard coded wait?